### ISSUE TITLE
Update seems to succeed, but ADODB throws an exception.

### ISSUE THREAD

**Issue by SDC-Armando** [ADOdb/ADOdb#887]
```
Fatal error:  Uncaught PDOException: SQLSTATE[IMSSP]: The active result for the query contains no fields. in C:\xampp\sdc\phplib\composer\vendor\adodb\adodb-php\drivers\adodb-pdo.inc.php:895
Stack trace:
#0 C:\xampp\sdc\phplib\composer\vendor\adodb\adodb-php\drivers\adodb-pdo.inc.php(895): PDOStatement->fetch(4)
#1 C:\xampp\sdc\phplib\composer\vendor\adodb\adodb-php\drivers\adodb-pdo.inc.php(818): ADORecordSet_pdo->_fetch()
#2 C:\xampp\sdc\phplib\composer\vendor\adodb\adodb-php\adodb.inc.php(1590): ADORecordSet_pdo->Init()
#3 C:\xampp\sdc\phplib\composer\vendor\adodb\adodb-php\adodb.inc.php(1527): ADOConnection->_Execute('\r\n    UPDATE\r\n ...', false)
#4 C:\xampp\sdc\appdir\_ajax\data_updateelectrologdata.php(24): ADOConnection->Execute('\r\n    UPDATE\r\n ...')
#5 {main}
  thrown in C:\xampp\sdc\phplib\composer\vendor\adodb\adodb-php\drivers\adodb-pdo.inc.php on line 895
```

The update, however, does succeed.

This is with Microsoft SQL Server 2019 using the PDO SQL Server driver for PHP.

https://learn.microsoft.com/en-us/sql/connect/php/download-drivers-php-sql-server?view=sql-server-ver16

**Comment by mnewnham** [comment]
I think that [this](https://stackoverflow.com/questions/16735898/the-active-result-contains-no-fields-using-pdo-with-ms-sql) is the answer you are looking for

**Comment by SDC-Armando** [comment]
> I think that [this](https://stackoverflow.com/questions/16735898/the-active-result-contains-no-fields-using-pdo-with-ms-sql) is the answer you are looking for

I actually tried the NOCOUNT and it didn't seem to fix the exception being thrown.

**Comment by SDC-Armando** [comment]
This is also happening on simple insert statements as well.

**Comment by SDC-Armando** [comment]
Moved to mssqlnative driver for inserts, deletes and updates - no errors.

**Comment by MrJackDuaner** [comment]
Hello,
I know this ticket is closed but I have the same error with PHP 8.1.13, ADODb 5.22.4 and PDO_SQLSRV 5.10.0, and I would like to understand why it doesn't work with ADODb.

My sample code :
**<?php
$adodbDir = 'ADOdb-5.22.4';

require $adodbDir . DIRECTORY_SEPARATOR . 'adodb.inc.php';
require $adodbDir . DIRECTORY_SEPARATOR . 'adodb-exceptions.inc.php';

$connectionString = 'sqlsrv:server=LOCALHOST;database=DATABASE;encrypt=true;trustServerCertificate=true';
$user = "USER";
$password = "PASSWORD";
$database = "DATABASE";
$query = "INSERT INTO TABLE (COL1, COL2) VALUES ('VAL1', 'VAL2')";

echo '<h2>Test with PDO</h2>';
$db = new PDO($connectionString, $user, $password);
try
{
	$db->exec($query);
	echo '<p>PDO OK</p>';
}
catch (PDOException $e) {
	echo '<p>PDO ERROR : ' . $e->GetMessage() . '</p>';
}

echo '<h2>Test with NewADOConnection</h2>';
$db = NewADOConnection('pdo');
$db->Connect($connectionString, $user, $password, $database);
try
{
	$db->Execute($query);
	echo '<p>NewADOConnection OK</p>';
}
catch (PDOException $e) {
	echo '<p>NewADOConnection ERROR : ' . $e->GetMessage() . '</p>';
}**

Result for PHP 8.1.13, ADODb 5.22.4 and PDO_SQLSRV 5.10.0 :
**Test with PDO
PDO OK
Test with NewADOConnection
NewADOConnection ERROR : SQLSTATE[IMSSP]: The active result for the query contains no fields.**

But if I switch to PHP 7.4.33...

Result for PHP 7.4.33, ADODb 5.22.4 and PDO_SQLSRV 5.10.0 :
**Test with PDO
PDO OK
Test with NewADOConnection
NewADOConnection OK**

Web server is IIS 10 and PHP is running as FastCGI. Both PHP configuration are similar: I always configure PHP in the same way.

If you need additional information or that I create a new ticket, please let me know.

Best regards.
